libs/wingc/src/files.rs

```diff
@@ -58,6 +58,10 @@ impl Files {
 		self.data.get(path.as_ref())
 	}
 
+	pub fn contains_file<S: AsRef<Path>>(&self, path: S) -> bool {
+		self.data.contains_key(path.as_ref())
```

Comment:
We unfortunately can't use canonicalize, but storing the inode makes sense.

Reply:
I don't think we can use inodes since they're not available on Windows

Reply:
As written, `Files` assumes each file is only indexed with a single, normalized path (to avoid issues of multiple ways to represent the same path) - so path normalization is the responsibility of the caller. Definitely something we can improve on, though I think I'll leave it for follow-up PRs.

Reply:
> I don't think we can use inodes since they're not available on Windows

While that's true, we should check to see what we actually get from WASI. It's very possible it's providing an equivalently useful number for us. I'm also cool with this being a followup issue

Reply:
👀  interesting. I'll create an issue

Reply:
> We unfortunately can't use canonicalize, but storing the inode makes sense.

Maybe delegating canonicalize to node through some WASM callback can solve this too. It'll be useful in other cases, like the JSII file caching where initially I used `canonicalize` to uniquely identify the .jsii file.